lib/include/openamp/rpmsg_virtio.h

```diff
@@ -73,6 +95,8 @@ struct rpmsg_virtio_device {
 	struct virtqueue *svq;
 	struct metal_io_region *shbuf_io;
 	struct rpmsg_virtio_shm_pool *shpool;
+	uint32_t desc_recycler_cnt;
+	struct desc_recycler_t desc_recycler[RPMSG_RECYCLER_SIZE];
```

Comment:
It's better to move them to rpmsg_virtio_shm_pool and add new api:
rpmsg_virtio_shm_pool_free_buffer or rpmsg_virtio_shm_pool_return_buffer

Reply:
The rpmsg_virtio_shm_pool structure is used to manage the shared memory for new buffer allocation. i would prefer in this case to create a new structure

Reply:
it's very common and natural that shm pool support the free function, like malloc is paired with free.